### PR TITLE
fix: add error message for purchasing duplicate baskets. rename skuError

### PR DESCRIPTION
**Pull request by christopappas**
If you have a 2 tabs open with the same basket, you place the order on 1 tab, then go to the _other_ tab and attempt to purchase, you will get a message in the stripe element saying "An unexpected error occurred" because the paymentIntent has already succeeded after placing the first order. The problem is that this message is not useful for the user.

Now, an error message will display instructing the user to reload their page, and once they reload, they will reach the empty basket page, which is the correct behavior given that they already purchased the basket.

## Checklist
- [ ] Consider PCI compliance impact and whether this PR changes how credit card information is handled.
- [ ] Intend to [release and monitor](https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories) this PR promptly after merge.

### CODE REVIEW

File: src/payment/checkout/payment-form/StripePaymentForm.jsx [hunk 1]
```diff
@@ -123,6 +123,8 @@ function StripePaymentForm({
         // Show error in payment form
         if (result.error.type === 'card_error' || result.error.type === 'validation_error') {
           setMessage(result.error.message);
+        } else if (result.error.type === 'invalid_request_error' && result.error.code === 'payment_intent_unexpected_state') {
```
Comment (julianajlk) [inline src/payment/checkout/payment-form/StripePaymentForm.jsx:126]: This if/else block is for showing a message to the user in the Stripe form (at the bottom). Is this what we want here on the new code added? If yes, can you update the comment on L123

File: src/payment/checkout/payment-form/StripePaymentForm.jsx [hunk 2]
```diff
@@ -154,7 +156,7 @@ function StripePaymentForm({
               // SDN failure: redirect to Ecommerce SDN error page.
               setLocation(`${getConfig().ECOMMERCE_BASE_URL}/payment/sdn/failure/`);
             } else if (errorData && errorData.sku_error) {
-              skuErrorDispatcher();
+              basketChangedErrorDispatcher();
```
Comment (julianajlk) [inline src/payment/checkout/payment-form/StripePaymentForm.jsx:159]: The variables were changed from "sku error" related to "basket changed", are you planning to keep the `sku_error` variable in ecommerce the same?

Comment (christopappas) [reply src/payment/checkout/payment-form/StripePaymentForm.jsx:159]: I am planning to leave sku_error as is in ecommerce, as a sku_error is just one type of error that indicates the situation where a basket has changed

File: src/payment/checkout/payment-form/StripePaymentForm.jsx (hunk 1 above)
Comment (christopappas) [reply src/payment/checkout/payment-form/StripePaymentForm.jsx:126]: `setMessage` will show the message in the Stripe element, whereas the saga dispatch in this if/else block will show a banner to the user. Would you like me to remove the comment?

**Pull request closed without merging**